deps/src/singular.cpp

```diff
@@ -5,6 +5,25 @@
 #include "ideals.h"
 #include "matrices.h"
 
+static std::string singular_return = std::string();
+static std::string singular_error = std::string();
+static std::string singular_warning = std::string();
+extern int         inerror;
```

Comment:
Perhaps add a comment here explaining that this is a variable from the Singular interpreter?